src/session_pool/session.js

```diff
@@ -305,7 +305,7 @@ export class Session {
      * @private
      */
     _puppeteerCookieToTough(puppeteerCookie) {
-        const isExpiresValid = puppeteerCookie.expires && typeof puppeteerCookie.expires === 'number';
+        const isExpiresValid = puppeteerCookie.expires && typeof puppeteerCookie.expires === 'number' && puppeteerCookie.expires !== -1;
```

Comment:
Would be better to check that it's non-negative, rather than exactly `-1`.

```js
const isExpiresValid = typeof puppeteerCookie.expires === 'number' && puppeteerCookie.expires > 0
```

Reply:
Also changing that, but in this case, I actually expect exactly `-1` as this is how puppeteer returns these cookies, if it's something else - it might be something completely different. On the other hand - other negative value would break it again, so yeah, better to have it more non-negative.